experiment/simulation/js/main.js

```diff
@@ -1 +1,377 @@
 //Your JavaScript goes in here
+
+document.addEventListener('DOMContentLoaded', function(){
+
+	const playButton = document.getElementById('play');
+	const pauseButton = document.getElementById('pause');
+	const restartButton = document.getElementById('restart');
+
+	pauseButton.addEventListener('click', function() { window.clearTimeout(tmHandle); });
+	playButton.addEventListener('click', function() { window.clearTimeout(tmHandle); tmHandle = setTimeout(draw, 1000 / fps); });
+	restartButton.addEventListener('click', function() {restart();});
+
+	function restart() 
+	{ 
+		window.clearTimeout(tmHandle); 
+
+		bldg = [
+			[[startx, height], [endx, height], [endx, 2 * height], [startx, 2 * height]],
+			[[startx, 2 * height], [endx, 2 * height], [endx, 3 * height], [startx, 3 * height]],
+			[[startx, 3 * height], [endx, 3 * height], [endx, 4 * height], [startx, 4 * height]]
+		];
+
+		ground = [
+			[startx - 300, 4 * height + 40],
+			[startx - 250, 4 * height - 40],
+			[endx + 300, 4 * height - 40],
+			[endx + 250 , 4 * height + 40],
+		];
+
+		a = -1 * m1 * m2 * m3;
+		b = m1 * m2 * k3 + m1 * m3 * (k2 + k3) + m2 * m3 * (k1 + k2);
+		c = k2 * k2 * m3 - m1 * k3 * (k3 + k2) - m2 * (k1 + k2) * k3 - m3 * (k1+ k2) * (k2 + k3) + m1 * k3 * k3;
+		d = (k1 + k2) * (k2 + k3) * (k3) - k2 * k2 * k3 - k3 * k3 * (k1 + k2);
+
+		lambda = CubicSolve(a, b, c, d);
+		lambda.sort();
+
+		if(mode == 1)
```

Comment:
Please use a descriptive enum for the mode types instead of bare numbers

Reply:
The value of the mode is self-descriptive as it shows the different modes of vibration of the system (i.e first, second and third modes of vibration). So using enum would be redundant.